src/OrchardCore/OrchardCore.Abstractions/Shell/Scope/ShellScope.cs

```diff
@@ -376,7 +376,19 @@ internal async Task ActivateShellInternalAsync()
         /// <summary>
         /// Registers a delegate to be invoked when 'BeforeDisposeAsync()' is called on this scope.
         /// </summary>
-        internal void BeforeDispose(Func<ShellScope, Task> callback) => (_beforeDispose ??= []).Insert(0, callback);
+        internal void BeforeDispose(Func<ShellScope, Task> callback, bool last)
+        {
+            var list = _beforeDispose ??= [];
```

Comment:
Better than a separate collection like I suggested.

Why the new local `list` though? and not just use `_beforeDispose` directly.

Reply:
Just personal habit...